### PR TITLE
Use std::apply

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: aten/src/ATen/core/ivalue_inl.h [hunk 1]
```diff
@@ -1175,9 +1175,9 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
   // set up before running the callback, as in, it will set up the CUDA streams,
   // synchronize them with the value, and so on (if needed).
   template<typename T>
-  void invokeCallback(T callback, bool uses_future) {
+  void invokeCallback(const T& callback, bool uses_future) {
```
Comment (Skylion007) [inline aten/src/ATen/core/ivalue_inl.h:1178]: I think this actually might break SFINAE unfortunately.

Comment (cyyever) [reply aten/src/ATen/core/ivalue_inl.h]: Use T&

**Pull request closed without merging**